### ISSUE TITLE
parser fails to recognize octals (#o prefix)

### ISSUE THREAD

**Issue by antifuchs** [talyz/fromElisp#3]
The following expression: `(set-file-modes server-socket-dir #o700)` fails to parse with the error `Unrecognized token on line 4694: #o700))`

I worked around by using read-from-string, but it would be nice to have these supported natively.

This was originally reported as https://github.com/nix-community/emacs-overlay/issues/164, but I guess emacs-overlay uses this package.

**Comment by league** [comment]
Yeah, I just encountered this with hexadecimal too, and traced it to here. It started with trying to do something like this in my init:

```elisp
(setq org-superstar-item-bullet-alist '((?* . ?•)))
```

But that's a unicode issue (unfortunate). I can use utf8 in comments and strings, but not for characters. So I tried replacing with the code point in hex:

```elisp
(setq org-superstar-item-bullet-alist '((?* . #x2022)))
```

and also not parsed. But the decimal version is okay.

```elisp
(setq org-superstar-item-bullet-alist '((?* . 8226)))
```
